### ISSUE TITLE
Enhancement: Click dynamic GridContextMenu

### ISSUE THREAD

**Issue by eiswind** [mvysny/karibu-testing#40]
How about integrating a helper for clicking a dynamic context menu?

I found a way with:

```
val tree = _get<TreeGrid<ProductTreeItem>>()
val collateralDetail = tree._get(2)
collateralDetail.name() shouldBe "CollateralDetail (Empty)"
val key = tree.dataCommunicator.keyMapper.key(collateralDetail)
val data = JreJsonFactory().parse<JsonObject>("{ \"event.detail\":{\"key\":\"${key}\"}}")
tree._fireDomEvent("vaadin-context-menu-before-open",data)
view.contextMenu._clickItemWithCaption("Add", collateralDetail)
```

**Comment by mvysny** [comment]
Hi, thank you for letting me know :+1: However I'm not sure what this code will solve - could you please provide a bit more details, e.g. what is the reason behind firing `vaadin-context-menu-before-open`, and in what kind of use-case this would be helpful?

**Comment by eiswind** [comment]
@mvysny the reasoning behind this is that a dynamic context menu needs to be initialized like

```
contextMenu= tree.addContextMenu()
contextMenu.setDynamicContentHandler { item ->
        if (item == null) {
            false
        } else {
            contextMenu.removeAll()
            item.fillContextMenu(contextMenu, tree)
        }
    }
```

firing the event does invoke the dynamicContentHandler, so the the MenuItems get initialized. Otherwise the menu would be empty.

```
fun <T> GridContextMenu<T>._clickDynamicContextMenuItemWithCaption(
        caption: String,
        tree: Grid<T>,
        item: T) {
    val key = tree.dataCommunicator.keyMapper.key(item)
    val data = JreJsonFactory().parse<JsonObject>("{ \"event.detail\":{\"key\":\"${key}\"}}")
    tree._fireDomEvent("vaadin-context-menu-before-open", data)
    this._clickItemWithCaption(caption, item)
}
```

**Comment by mvysny** [comment]
Excellent idea, thank you :+1: I'm even thinking of always firing the `vaadin-context-menu-before-open` DOM event from _clickItemWithCaption, in order to simulate the Vaadin initialization more closely. I'll prototype this.

**Comment by mvysny** [comment]
Actually the easier way to invoke the DynamicContentHandler is to call it directly via contextMenu.getDynamicContentHandler(). I've modified the `_clickItemWithCaption()` functions to call the DynamicContentHandler directly.

**Comment by eiswind** [comment]
Looks great!

**Comment by mvysny** [comment]
Excellent, thanks for letting me know :) Released as Karibu-Testing 1.1.27
